### ISSUE TITLE
Queues, Exchanges, Bindings are created in wrong brokers.

### ISSUE THREAD

**Issue by misterflud** [freenowtech/spring-multirabbit#22]
Hi

Problems with Queues, Exchanges, Bindings which have been declare by annotation `@Bean `like this:
`    
@Bean
    public Queue myQueue() {
        return QueueBuilder
                .durable("QueueName")
                .build();
    }
`

MultiRabbitListenerAnnotationBeanPostProcessor method (enhanceBeansWithReferenceToRabbitAdmin) takes all Beans from applicationContext instead only Bean which can be created from an annotation `@RabbitListener` and of course all Queues (which have been declared by annotation `@Bean`) move to first created RabbitAdmin (multi or default) and after that they crate in wrong broker... 

I think the library should manage:
1) Queus are crated by `@Bean`
2) Queus are crated by by annotation `@Queue`

Thank you for your attention

**Comment by abedurftig** [comment]
Hey, I just worked through the same. The solution I have found looks like this.

1) Inject the `RabitAdmin` for your connection.
```
@Configuration
class RabbitConfiguration(
    @Qualifier("connectionA-admin") private val connectionARabbitAdmin: RabbitAdmin
) {
```
2) Declare the queues, exchanges and bindings programmatically.

```
@PostConstruct
fun initRabbit() {
    connectionARabbitAdmin.declareExchange(deadLetterExchange())
    connectionARabbitAdmin.declareExchange(exchange())
    connectionARabbitAdmin.declareQueue(deadLetterQueue())
    connectionARabbitAdmin.declareQueue(queue())
    connectionARabbitAdmin.declareBinding(deadLetterQueueBinding())
    connectionARabbitAdmin.declareBinding(binding())
}

private fun deadLetterExchange(): DirectExchange {
    return DirectExchange(DLQ_EXCHANGE_NAME)
}
```

**Comment by rwanderc** [comment]
Thx @misterflud for the issue and @abedurftig for the suggestion.
There is also another possible solution... you can define the `RabbitAdmin` that will be responsible for managing the `Declarable` directly in the bean you are creating.
For example:
```
    @Bean
    org.springframework.amqp.core.Queue someQueue() {
        final org.springframework.amqp.core.Queue queue = QueueBuilder.durable("someQueue").build();
        queue.setAdminsThatShouldDeclare("connectionA-admin");
        return queue;
    }
```

This way, MultiRabbit will skip from assigning it automatically to a `RabbitAdmin`.
And actually, the default behavior from `spring-amqp` (without MultiRabbit) is that `Declarables` are declared in all `RabbitAdmin`s if the `setAdminsThatShouldDeclare()` is not specified. MultiRabbit assumes a diff behavior by assigning them to the default broker.

**Comment by rwanderc** [comment]
This fix is possible will be possible in version v2.4.0, since `spring-amqp` now offers a way to find out the declarables created, and therefore avoid processing the individual declarable beans.

**Comment by rwanderc** [comment]
@misterflud this should be fixed since Spring MultiRabbit 2.4.
Please feel free to re-open the issue if you still face the same problem.
